### DISCUSSION TITLE
prediction locations using mp model

### DISCUSSION THREAD

**Discussion by Spines151** [ianjonsen/foieGras#70] [Q&A]
Hello 
I have a question relating to predicting locations.
I have tag data (ARGOS and GPS) from North Atlantic Right Whales in the Gulf of St. Lawrence. Mean/median time between subsequent positions is 40/22min. I fitted rw, crw, mp models with different time steps and max speed. The osar residuals for the ‘mp’ (1-hour time step, 2m/s max speed) model looked ‘best’. The predicted locations fit the raw GPS data well. 
I have also 14,000 dives, and I wanted to predict their locations and, in a further step, add the information to inform an hMM to help distinguish between the behavioural states. But first, I wanted to plot them. I used the best-performing model ‘mp’ and the data frame with the start times for every dive. However, while most dives fall somewhat within the range of the path of the mp model (using 1-hour predictions), I have some dives dozens or even hundreds of km away, further away than even the most extreme ARGOS position (and should not be possible given the 2m/s max speed). I repeated the same exercise with the ‘crw’ model, and it worked. The dives were all very close to the path of the 1-hour prediction of the same model. I do not understand why. I can simply use the ‘crw’ model, but the residuals of the ‘mp’ model looked better, although, in the end, I will likely not use the estimated move persistence of the mp model. 
Many thanks for any suggestions.
Chris

**Comment by ianjonsen** [comment]
Hi,

Based on your description, there are multiple interpretations of how you actually coded this using `aniMotum` functions. In order for me to understand exactly what you've done and provide a solution, I would need to see both your code and a small reproducible example.

Thanks, Ian

**Reply by Spines151** [reply]
Hey Ian,
Many thanks for the fast response. I had issues replicating the same error. Somehow, when I run the model for a single tag and not as a batch, the extreme outliers (predictions) are gone. Still, the main question remains: why are the predicted locations for the ‘mp’ model (when using the dive times as time step) considerably outside the path for the predicted model (using a regular 1 hrs time step)? I have attached the code and the files (location and dive times as separated data frames) for two tags. One animal (tag 158387) is moving larger distances and staying in some areas; the second one (197192) is not moving much at all. It seems the issue is minimal for the first and worse for the latter. 
I re-run the ‘crw’ model, and it is now very similar, the predictions using the dive times as time step are off the ones using the 1-hr time step. The second tag (197192) does not fit well. 
I used QGIS to plot the predictions. 
Many thanks again
Chris
[divetime_158387.csv](https://github.com/user-attachments/files/18065546/divetime_158387.csv)
[divetime_197192.csv](https://github.com/user-attachments/files/18065547/divetime_197192.csv)
[narw_158387.csv](https://github.com/user-attachments/files/18065548/narw_158387.csv)
[narw_197192.csv](https://github.com/user-attachments/files/18065549/narw_197192.csv)

#example NARW SSM model with dive time

#setting directory
#setwd("~/NARW SSM/")

#Loading packages
require(tidyverse)
require(ggspatial)
require(patchwork)
require(sf)
require(aniMotum)
require(openxlsx) 

# read in tag data
#example 1st animal with large movements and intertwined movements

narw_158387 <- read.csv("narw_158387.csv")

narw_158387$date <- as.POSIXct(narw_158387$date, format="%Y-%m-%d %H:%M:%S", tz="UTC")

narw_158387$id <-as.character(narw_158387$id)

#example 2nd animal only conducting intertwined movements

narw_197192 <- read.csv("narw_197192.csv")

narw_197192$date <- as.POSIXct(narw_197192$date, format="%Y-%m-%d %H:%M:%S", tz="UTC")

narw_197192$id <-as.character(narw_197192$id)

#example 1
# read in dive data and convert id to character and date_time as.POSIXct
dd_158387 <-read.csv("divetime_158387.csv")

dd_158387$date <- as.POSIXct(dd_158387$date, format="%Y-%m-%d %H:%M:%S", tz="UTC")

dd_158387$id <-as.character(dd_158387$id)

#example 2

dd_197192 <-read.csv("divetime_197192.csv")

dd_197192$date <- as.POSIXct(dd_197192$date, format="%Y-%m-%d %H:%M:%S", tz="UTC")

dd_197192$id <-as.character(dd_197192$id)


# fit SSM model, 'mp', 1 hour time step, max speed 2 m/s  

#example 1

fit_158387_mp_1hr_2ms <- fit_ssm(narw_158387, vmax = 2, model = 'mp', time.step = 1, map = list(psi=factor(NA)))

fit_158387_mp_dd_2ms <- fit_ssm(narw_158387, vmax = 2, model = 'mp', time.step = dd_158387, map = list(psi=factor(NA)))

predicted_158387_mp_1hr_2ms <-grab(fit_158387_mp_1hr_2ms, what = "p", normalise = FALSE)
write.table(predicted_158387_mp_1hr_2ms, file = paste0("Predicted_tag_158387_mp_1hr_2ms_not_normalised.csv"), row.names = F, sep = ",", dec=".", fileEncoding = "latin1")

predicted_158387_mp_dd_2ms <-grab(fit_158387_mp_dd_2ms, what = "p", normalise = FALSE)
write.table(predicted_158387_mp_dd_2ms, file = paste0("Predicted_tag_158387_mp_dive_times_2ms_not_normalised.csv"), row.names = F, sep = ",", dec=".", fileEncoding = "latin1")


#example 2

fit_197192_mp_1hr_2ms <- fit_ssm(narw_197192, vmax = 2, model = 'mp', time.step = 1, map = list(psi=factor(NA)))

fit_197192_mp_dd_2ms <- fit_ssm(narw_197192, vmax = 2, model = 'mp', time.step = dd_197192, map = list(psi=factor(NA)))

predicted_197192_mp_1hr_2ms <-grab(fit_197192_mp_1hr_2ms, what = "p", normalise = FALSE)
write.table(predicted_197192_mp_1hr_2ms, file = paste0("Predicted_tag_197192_mp_1hr_2ms_not_normalised.csv"), row.names = F, sep = ",", dec=".", fileEncoding = "latin1")

predicted_197192_mp_dd_2ms <-grab(fit_197192_mp_dd_2ms, what = "p", normalise = FALSE)
write.table(predicted_197192_mp_dd_2ms, file = paste0("Predicted_tag_197192_mp_dive_times_2ms_not_normalised.csv"), row.names = F, sep = ",", dec=".", fileEncoding = "latin1")
